### PR TITLE
Add Updating klusterlet-addon-search on manged clusters

### PR DESCRIPTION
**Pull request by dockerymick**
Also added the known issue 

https://github.com/stolostron/backlog/issues/27109
https://github.com/stolostron/backlog/issues/27173

Deciding not to do arch changes such as creating new files and changing anchors

### CODE REVIEW

File: observability/search.adoc [hunk 1]
```diff
@@ -40,18 +50,84 @@ metadata:
...
+spec:
+  dbConfig: tuning-config
+  dbStorage:
+  size: 10Gi
+  storageClassName: gp2-search
+----
++
+- Optimize cost by tuning the pod memory requirements, replica count, and update log levels for active pods. Update the `deployment` section of the `searchcustomization` custom resource. There are four deployments managed by the `searchcustomization`, which can be updated individually. Your `searchcustomization` custom resource might resemble the following file:
++
+[source,yaml]
```
Comment (xrajesh) [inline observability/search.adoc]: @dockerymick F06 2.6 line 60 to 88 is not valid.

File: observability/search.adoc [hunk 2]
```diff
@@ -40,18 +50,84 @@ metadata:
...
+      collector:
+      resources:
+        limits:
+          cpu: 500m
+          memory: 128Mi
+        requests:
+          cpu: 250m
+          memory: 64Mi
+        indexer:
+        replicaCount: 3
```
Comment (xrajesh) [inline observability/search.adoc]: @dockerymick line 90 to 107 is not valid for release 2.6

File: observability/search.adoc [hunk 3]
```diff
@@ -40,18 +50,84 @@ metadata:
...
+  database: {}
+  indexer: {}
+  queryapi: {}
+ nodeSelector:
+  node-role.kubernetes.io/infra: ""
+ tolerations:
+ - effect: NoSchedule
+  key: node-role.kubernetes.io/infra
+  operator: Exists
+----
```
Comment (xrajesh) [inline observability/search.adoc:53]: @dockerymick line 110 to 130 is not valid for 2.6

File: observability/search.adoc [hunk 4]
```diff
@@ -40,18 +50,84 @@ metadata:
...
 spec:
-  persistence: true
-  storageClass: gp2
-  storageSize: 12Gi
+  dbStorage:
+  size: 10Gi
+  storageClassName: gp2
 ----
++
+This confirguration creates a PVC named `gp2-search` and is mounted to the `search-postgres` pod. By default, the storage size is `10Gi`. You can modify the storage size. For example, `20Gi` might be sufficient for about 200 managed clusters.
```
Comment (xrajesh) [inline observability/search.adoc]: For ACM 2.6 line 53 to 58 is not valid @dockerymick

File: observability/search.adoc [hunk 5]
```diff
@@ -6,30 +6,40 @@ For {product-title}, search provides visibility into your Kubernetes resources a
...
+----
+oc get search searchcustomization -o yaml
+----
 
+The search operator watches the `searchcustomization` custom resource, reconciles the changes and updates active pods. View the following descriptions of the configurations:
+
+- PostgreSQL database storage: 
++
+When you install {product-title-short}, the PostgreSQL database is configured to save the PostgreSQL data in an empty directory (`emptyDir`) volume. If the empty directory size is limited, you can save the PostgreSQL data on a Persistent Volume Claim (PVC) to improve search performance. You can select a storageclass from your {product-title-short} hub cluster to back up your search data. For example, if you select the `gp2` storageclass your configuration might resemble the following example:
++
```
Comment (xrajesh) [inline observability/search.adoc]: For ACM 2.6 line 33 to 42 are not valid @dockerymick

File: observability/search.adoc [hunk 6]
```diff
@@ -6,30 +6,40 @@ For {product-title}, search provides visibility into your Kubernetes resources a
...
 
-* *Search API*: Provides access to the data in the search index and enforces role-based access control. 
+- `search-api`: Provides access to all cluster data in the `search-indexer` through GraphQL and enforces role-based access control (RBAC).
 
-Search is enabled by default. Search is also enabled when you provision or manually import a managed cluster. If you want to disable search on your managed cluster, see link:../clusters/modify_endpoint.adoc#modifying-the-klusterlet-add-ons-settings-of-your-cluster[Modifying the klusterlet add-ons settings of your cluster] for more information.
+- `search-postgres`: Stores collected data from all managed clusters in an instance of the PostgreSQL database.
+Search is configured by default on the hub cluster. When you provision or manually import a managed cluster, the `klusterlet-addon-search` is enabled. If you want to disable search on your managed cluster, see link:../add-ons/modify_endpoint.adoc#modifying-the-klusterlet-add-ons-settings-of-your-cluster[Modifying the klusterlet add-ons settings of your cluster] for more information.
```
Comment (xrajesh) [inline observability/search.adoc:26]: @dockerymick For ACM 2.6 , please dont make above line ADDs you have, from 18 to 27 . Leave it as is .

File: observability/search.adoc (hunk 1 above)
Comment (dockerymick) [reply observability/search.adoc]: updated

File: observability/search.adoc (hunk 2 above)
Comment (dockerymick) [reply observability/search.adoc]: updated

File: observability/search.adoc (hunk 3 above)
Comment (dockerymick) [reply observability/search.adoc:53]: updated

File: observability/search.adoc (hunk 4 above)
Comment (dockerymick) [reply observability/search.adoc]: updated

File: observability/search.adoc (hunk 5 above)
Comment (dockerymick) [reply observability/search.adoc]: updated

File: observability/search.adoc (hunk 6 above)
Comment (dockerymick) [reply observability/search.adoc:26]: updated

**Comment by dockerymick** [comment]
Thanks for your review! I have addressed your comments. When you have an opportunity, will you provide your approval? @xrajesh

File: observability/search.adoc [hunk 7]
```diff
@@ -6,53 +6,31 @@ For {product-title}, search provides visibility into your Kubernetes resources a
...
 [#search-components]
 == Search components
 
 The search architecture is composed of the following components:
 
-* *Collector*: Watches the Kubernetes resources and creates an index. The `search-collector` computes relationships for resources within the managed cluster.
+- `search-collector`: Watches the Kubernetes resources, collects the resource metadata, computes relationships for resources across all of your managed clusters, and sends the collected data to the `search-indexer`. The `search-collector` on your managed cluster runs as a pod named, `klusterlet-addon-search`. 
 
-* *Aggregator*: Receives data from the collector and writes to a database. The `search-aggregator` watches resources in the hub cluster, computes multicluster relationships, and tracks the activity from connected collectors.
+- `search-indexer`: Receives resource metadata from the collectors and writes to PostgreSQL database. The `search-indexer` also watches resources in the hub cluster to keep track of active managed clusters.
```
Comment (xrajesh) [inline observability/search.adoc]: @dockerymick This line need not change for 2.6 , please dont update it.

File: observability/search.adoc [hunk 8]
```diff
@@ -6,53 +6,31 @@ For {product-title}, search provides visibility into your Kubernetes resources a
...
 
-* *Search API*: Provides access to the data in the search index and enforces role-based access control. 
-
-Search is enabled by default. Search is also enabled when you provision or manually import a managed cluster. If you want to disable search on your managed cluster, see link:../clusters/modify_endpoint.adoc#modifying-the-klusterlet-add-ons-settings-of-your-cluster[Modifying the klusterlet add-ons settings of your cluster] for more information.
+- `search-api`: Provides access to all cluster data in the `search-indexer` through GraphQL and enforces role-based access control (RBAC).
```
Comment (xrajesh) [inline observability/search.adoc]: @dockerymick This line need not change for 2.6 , please dont update it.

File: observability/search.adoc [hunk 9]
```diff
@@ -6,53 +6,31 @@ For {product-title}, search provides visibility into your Kubernetes resources a
...
 
 [#search-customization]
 == Search customization
 
-When you install {product-title-short}, the product is configured to persist the data in-memory to a file system. The StatefulSet `search-redisgraph` deploys the Redisgraph pod, which mounts the persistent volume named `persist`. If your cluster has a defined default storage class, the search component creates a Persistent Volume Claim (PVC) of 10Gi on the default storage class. If a default storage class does not exist in your cluster, search saves the index in an empty directory (`emptyDir`).
-
-You can customize the storage settings for search by creating the `searchcustomization` CR. Search customization is namespace-scoped and located where search is installed in the hub cluster. View the following example of the search customization CR:
+You can modify the default values in the `searchcustomization` custom resource. To view details of the custom resource, run the following command:
```
Comment (xrajesh) [inline observability/search.adoc]: @dockerymick This line need not change for 2.6 , please dont update it.

File: observability/search.adoc [hunk 10]
```diff
@@ -6,53 +6,31 @@ For {product-title}, search provides visibility into your Kubernetes resources a
...
-metadata:
-  name: searchcustomization
-  namespace: open-cluster-management
-  labels:
-    cluster.open-cluster-management.io/backup: ""
-spec:
-  persistence: true
-  storageClass: gp2
-  storageSize: 12Gi
+oc get search searchcustomization -o yaml
```
Comment (xrajesh) [inline observability/search.adoc]: @dockerymick This line need not change for 2.6 , please dont update it.

File: observability/search.adoc [hunk 11]
```diff
@@ -6,53 +6,31 @@ For {product-title}, search provides visibility into your Kubernetes resources a
...
 ----
 
-Run the following command to view search customization CRD: 
-
-----
-oc get crd searchcustomizations.search.open-cluster-management.io -o yaml
-----
-
-You can disable persistence by updating the `persistence` flag to `false` in the customization CR, which turns off saving search index to the file system. A status for persistence can be retrieved from the search operator (`searchoperator`) CR. Run the following command to view search operator CR: `oc get searchoperator searchoperator -o yaml`. 
-
```
Comment (xrajesh) [inline observability/search.adoc:55]: @dockerymick This line need not change for 2.6 , please dont update it.

File: observability/search.adoc (hunk 7 above)
Comment (dockerymick) [reply observability/search.adoc]: @xrajesh are you requesting that I revert the change back to the following information:


```
* *Aggregator*: Receives data from the collector and writes to a database. The `search-aggregator` watches resources in the hub cluster, computes multicluster relationships, and tracks the activity from connected collectors.
```

Comment (xrajesh) [reply observability/search.adoc]: Correct @dockerymick

Comment (dockerymick) [reply observability/search.adoc]: updated

File: observability/search.adoc (hunk 11 above)
Comment (dockerymick) [reply observability/search.adoc:55]: updated

**Comment by dockerymick** [comment]
@xrajesh thanks for your feedback and patience. I have reverted the changes that you requested. Let me know if anything else needs to change. In support of being on train 1, I will not update 2.5 unless it is customer reported

File: observability/search.adoc (hunk 9 above)
Comment (dockerymick) [reply observability/search.adoc]: Updated

File: observability/search.adoc (hunk 10 above)
Comment (dockerymick) [reply observability/search.adoc]: Updated

**Review by xrajesh — APPROVED** [review]

**Comment by dockerymick** [comment]
@xrajesh thanks for your approval! Merging this PR

**Comment by dockerymick** [comment]
https://issues.redhat.com/browse/ACM-3365

**Pull request merged**